tests/integration_tests.rs

```diff
@@ -1012,3 +1032,62 @@ async fn test_pod_mounts_with_pvc() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+const RESOURCE_NAME: &str = "example.com/gpu";
+const RESOURCE_ENDPOINT: &str = "gpu-device-plugin.sock";
+
+async fn create_device_plugin_resource_pod(
+    client: kube::Client,
+    pods: &Api<Pod>,
+    resource_manager: &mut TestResourceManager,
+) -> anyhow::Result<()> {
+    use k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+    let containers = vec![
+        WasmerciserContainerSpec::named("device-plugin-test").with_args(&[
+            "write(lit:watermelon)to(stm:stdout)",
+            "assert_exists(env:DEVICE_PLUGIN_VAR)",
+        ]),
+    ];
+
+    let mut requests = std::collections::BTreeMap::new();
+    requests.insert(RESOURCE_NAME.to_string(), Quantity("1".to_string()));
+    let resources = ResourceRequirements {
+        limits: requests.clone(),
+        requests: requests,
+    };
+
+    wasmercise_wasi(
+        DEVICE_PLUGIN_RESOURCE_POD,
+        client,
+        pods,
+        vec![],
+        containers,
+        vec![],
+        Some(resources),
+        OnFailure::Panic,
+        resource_manager,
+    )
+    .await
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_pod_with_device_plugin_resource() -> anyhow::Result<()> {
+    println!("Starting DP test");
+    let test_ns = "wasi-e2e-pod-with-device-plugin-resource";
+    let (client, pods, mut resource_manager) = set_up_test(test_ns).await?;
+
+    device_plugin::launch_device_plugin(RESOURCE_NAME, RESOURCE_ENDPOINT).await?;
+
+    // Create a Pod that requests the DP's resource
+    create_device_plugin_resource_pod(client.clone(), &pods, &mut resource_manager).await?;
+    assert::pod_exited_successfully(&pods, DEVICE_PLUGIN_RESOURCE_POD).await?;
+    assert::pod_container_log_contains(
```

Comment:
Yeah, I think we can drop this for now as we get all we need from just checking if the pod exited successfully

Reply:
Good point! Removed ce2fcb4531488cd7f4fa7086b9add02d36b21c5e